vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -469,19 +467,23 @@ public void finish() {
       GameModule.getGameModule().warn(Resources.getString("PlayerRoster.failed_pref_write", e.getLocalizedMessage()));
     }
 
-    final String newSide = untranslateSide(sideConfig.getValueString());
+   // Drop into standard routine, starting with checking side is still available (race condition mitigation)
+    // returns untranslated side
+    final String newSide = promptForSide(sideConfig.getValueString());
+
     if (newSide != null) {
       if (GameModule.getGameModule().isMultiplayerConnected()) {
-        final Command c = new Chatter.DisplayText(GameModule.getGameModule().getChatter(), Resources.getString(GlobalOptions.getInstance().chatterHTMLSupport() ? "PlayerRoster.joined_side_2" : "PlayerRoster.joined_side", GameModule.getGameModule().getPrefs().getValue(GameModule.REAL_NAME), newSide));
+        final Command c = new Chatter.DisplayText(GameModule.getGameModule().getChatter(), Resources.getString(GlobalOptions.getInstance().chatterHTMLSupport() ? "PlayerRoster.joined_side_2" : "PlayerRoster.joined_side", GameModule.getGameModule().getPrefs().getValue(GameModule.REAL_NAME), translateSide(newSide)));
         c.execute();
       }
 
       final Add a = new Add(this, GameModule.getActiveUserId(), GlobalOptions.getInstance().getPlayerId(), newSide);
       a.execute();
       GameModule.getGameModule().getServer().sendToOthers(a);
+
+      getLaunchButton().setVisible(getMySide() != null);
```

Comment:
Is this the Retire button? Should this be calling `setEnable` now?

Reply:
Yes, I need to adjust this PR.  Also, I am considering whether setVisibility should still have a role for when no sides are defined.